test/conformance/header_test.go

```diff
@@ -83,22 +84,25 @@ func TestShouldHaveHeadersSet(t *testing.T) {
 	headers := ri.Request.Headers
 
 	for header, regex := range expectedHeaders {
-		hv, ok := headers[header]
-		if !ok {
-			t.Errorf("Header %s was not present on request", header)
-			continue
-		}
+		hvl, ok := headers[http.CanonicalHeaderKey(header)]
+		// Check against each value for the header key
+		for _, hv := range hvl {
+			if !ok {
```

Comment:
I think you want that `if` above the loop.

Reply:
Whoops. Fixed.